### ISSUE TITLE
Add a rehydrate function 

### ISSUE THREAD

**Issue by kof** [cssinjs/jss#677]
Initially I went against creating a separate rehydrate function, because we decided to not actually do rehydration, but instead just regenerate the styles, because it is fast enough and more robust. 

So if I did `jss.rehydrate('.style-selector')´ it would be basically inserting the styles after this element and just removing the old style node.

Now as some people had issues with mismatching selectors from SSR, it makes sense to add some more logic to rehydrate, which will make sure in dev mode that user gets proper error report.

1. Check the JSS versions matching between ssr and client.
1. Check the amount of JSS instances matches between ssr and client.
1. Check the module id matches between ssr and client.

All checks could be done for e.g. if ssr generates a comment string which contains the information (dev mode only). Rehydration call checks the comment and prints a warning or error.

**Comment by kof** [comment]
cc @jedwards1211

**Comment by Aetherall** [comment]
Indeed, It could be really usefull to have those error reports. Actually managing packages which uses jss as dependency is a pain
